src/Shared/FileMatcher.cs

```diff
@@ -2499,13 +2499,13 @@ private static string[] CreateArrayWithSingleItemIfNotExcluded(string filespecUn
             catch (AggregateException ex)
             {
                 // Flatten to get exceptions than are thrown inside a nested Parallel.ForEach
-                if (ex.Flatten().InnerExceptions.All(ExceptionHandling.IsIoRelatedException))
+                if (ex.Flatten().InnerExceptions.All(ex => ExceptionHandling.IsIoRelatedException(ex) && !(ex is FileLoadException)))
```

Comment:
nit:
move this up to
`catch (AggregateException ex) when...`
?

Reply:
That won't be equivalent. The `ex` here is different from the `ex` being caught on line 2499. Suggestion: rename the lambda parameter.

Reply:
Apologies, I misunderstood. Moving everything to an exception filter is a great idea!

Reply:
Decided that it is better to rethrow the FileLoadException as a InvalidOperationException directly in ManagedFileSystem class and not here. It will allow to throw a proper error if enumeration is used elsewhere besides FileMatcher class  some time in future. So, no need for this line anymore, but thank you for your suggestions!